armbian/base/scripts/autosetup-ssd.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+set -eu
+
+# BitBox Base: auto-setup of SSD
+#
+
+function usage() {
+    printf "
+BitBox Base: Auto-Setup SSD
+
+This script checks for potential storage targets and automates the setup process.
+Use with caution, data on targets will be deleted.
+
+Usage: $0 <status|format> [device] [--assume-yes]
+
+Examples:
+* $0 status                     Get list of potential target devices in system
+* $0 format /dev/sda            Set up specified device interactively
+* $0 format auto                Detect device and set up interactively
+* $0 format auto --assume-yes   WARNING! Execute only in well defined environments!
+
+"
+}
+
+function list_targets() {
+    echo
+    printf "%-16s %-7s %16s    %-30s\n" "DEVICE" "FSTYPE" "SIZE" "OK for storage?"
+    printf "%-16s %-7s %16s   %-30s\n" "---------------" "------" "-----------------" "------------------------------"
+
+    device_found=0
```

Comment:
I'd suggest explicitly adding `local` when defining local variables, since otherwise they are still globally accessible, even if defined inside a function. I.e let's do:
```
local device_found=0
local targets_total=0
local blockdev_target_path=""
```

Reply:
These variables are actually used outside the function as well. As we've spent too much time already on this script that's not necessary for production and is more for development, I'll leave that for a future iteration.